app/questionnaire/navigator.py

```diff
@@ -303,3 +303,56 @@ def get_latest_location(self, completed_blocks=None):
             'group_id': SchemaHelper.get_first_group_id(self.survey_json),
             'group_instance': 0,
         }
+
+    def get_front_end_navigation(self, group_id, group_instance, completed_blocks):
+
+        navigation = []
+
+        for group in self.survey_json['groups']:
+
+            repeating_rule = self.get_repeating_rule(group)
+            last_questionnaire_block = self.get_last_questionnaire_block_in_group(group)
+
+            # We only want to show groups that have questionnaire blocks in, interstitial groups are excluded
+            if last_questionnaire_block:
+
+                if repeating_rule:
+                    no_of_repeats = evaluate_repeat(repeating_rule, self.answer_store)
+                    answer_id = repeating_rule['answer_id']
+                    answers = self.answer_store.filter(answer_id=answer_id)
+
+                    for i in range(no_of_repeats):
+                        if answers[i]['value']:
+                            navigation.append({
+                                'link_name': answers[i]['value'],
+                                'path': str(group['id']) + '/' + str(i) + '/' + str(group['blocks'][0]['id']),
```

Comment:
Having had a quick discussion here, it possibly would be better to make the call to an url generator (which you'd need to define) within the templates, using a method which only takes group_id, instance and block. This would save having to define the path at all within navigator.